### PR TITLE
Add support for code metadata in mandos

### PR DESCRIPTION
**Pull request by CostinCarabas**

### CODE REVIEW

File: scenarioexec/util.go [hunk 1]
```diff
@@ -42,6 +42,7 @@ func convertAccount(testAcct *mj.Account, world *worldmock.MockWorld) (*worldmoc
 		AsyncCallData:   testAcct.AsyncCallData,
 		ShardID:         uint32(testAcct.Shard.Value),
 		IsSmartContract: len(testAcct.Code.Value) > 0,
+		// CodeMetadata:    testAcct.CodeMetadata.Value,
```
Comment (CostinCarabas) [inline scenarioexec/util.go:45]: TODO: Remove commented Code

File: scenarioexec/stepCheckAccounts.go [hunk 2]
```diff
@@ -99,6 +99,16 @@ func (ae *VMTestExecutor) checkAccounts(baseErrMsg string, checkAccounts *mj.Che
...
 		}
 
+		if !expectedAcct.CodeMetadata.IsUnspecified() && !expectedAcct.CodeMetadata.Check(matchingAcct.CodeMetadata) {
+			return fmt.Errorf("%s bad account codeMetadata. Account: %s. Want: %s. Have: \"%s\"",
+				baseErrMsg,
+				expectedAcct.Address.Original,
+				oj.JSONString(expectedAcct.CodeMetadata.Original),
+				ae.exprReconstructor.Reconstruct(
+					matchingAcct.CodeMetadata,
+					er.CodeHint)) // TODO: Metadata hint?
```
Comment (CostinCarabas) [inline scenarioexec/stepCheckAccounts.go:109]: How to approach Metadata hint?

Comment (andrei-marinica) [reply scenarioexec/stepCheckAccounts.go:109]: Yes, it's appropriate to have a codeMetadata hint
It should always decode as hex.

Comment (andrei-marinica) [reply scenarioexec/stepCheckAccounts.go:109]: Also NoHint works for now

File: scenarioexec/util.go (hunk 1 above)
Comment (CostinCarabas) [reply scenarioexec/util.go:45]: In the following lines (46-50) we hardcode the metadata to have all permissions by default. This is needed so we don't modify all the tests.

File: test/features/composability/scenarios/forw_raw_contract_deploy.scen.json [hunk 3]
```diff
@@ -117,14 +117,14 @@
...
-                    "str:some_argument"
+                    "0x0502"
                 ],
                 "gasLimit": "50,000,000",
                 "gasPrice": "0"
             },
             "expect": {
                 "out": [
-                    "str:some_argument",
+                    "0x0502",
```
Comment (CostinCarabas) [inline test/features/composability/scenarios/forw_raw_contract_deploy.scen.json:127]: @andrei-marinica , 
Test for deploy from source.

**Pull request closed without merging**